schema-registry/src/main/java/io/streamnative/pulsar/handlers/kop/schemaregistry/HttpRequestProcessor.java

```diff
@@ -0,0 +1,138 @@
+ */
+package io.streamnative.pulsar.handlers.kop.schemaregistry;
+
+import static io.netty.handler.codec.http.HttpResponseStatus.INTERNAL_SERVER_ERROR;
+import static io.netty.handler.codec.http.HttpResponseStatus.NO_CONTENT;
+import static io.netty.handler.codec.http.HttpResponseStatus.OK;
+import static io.netty.handler.codec.http.HttpVersion.HTTP_1_1;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.SerializationFeature;
+import io.netty.buffer.Unpooled;
+import io.netty.handler.codec.http.DefaultFullHttpResponse;
+import io.netty.handler.codec.http.FullHttpRequest;
+import io.netty.handler.codec.http.FullHttpResponse;
+import io.netty.handler.codec.http.HttpHeaderNames;
+import io.netty.handler.codec.http.HttpResponseStatus;
+import io.netty.util.CharsetUtil;
+import io.streamnative.pulsar.handlers.kop.schemaregistry.model.impl.SchemaStorageException;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionException;
+import lombok.AllArgsConstructor;
+
+public abstract class HttpRequestProcessor implements AutoCloseable {
+
+    protected static final ObjectMapper MAPPER = new ObjectMapper()
+            .configure(SerializationFeature.INDENT_OUTPUT, true);
+
+    public static FullHttpResponse buildStringResponse(String body, String contentType) {
+        FullHttpResponse httpResponse = new DefaultFullHttpResponse(HTTP_1_1, OK,
+                Unpooled.copiedBuffer(body, CharsetUtil.UTF_8));
+        httpResponse.headers().set(HttpHeaderNames.CONTENT_TYPE, contentType);
+        httpResponse.headers().set(HttpHeaderNames.CONTENT_LENGTH, body.length());
+        addCORSHeaders(httpResponse);
+        return httpResponse;
+    }
+
+    public static FullHttpResponse buildEmptyResponseNoContentResponse() {
+        FullHttpResponse httpResponse = new DefaultFullHttpResponse(HTTP_1_1, NO_CONTENT, Unpooled.EMPTY_BUFFER);
+        httpResponse.headers().set(HttpHeaderNames.ALLOW, "GET, POST, PUT, DELETE");
+        httpResponse.headers().set(HttpHeaderNames.CONTENT_LENGTH, 0);
+        addCORSHeaders(httpResponse);
+        return httpResponse;
+    }
+
+    public static FullHttpResponse buildErrorResponse(HttpResponseStatus error, String body, String contentType) {
+        FullHttpResponse httpResponse = new DefaultFullHttpResponse(HTTP_1_1, error,
+                Unpooled.copiedBuffer(body, CharsetUtil.UTF_8));
+        httpResponse.headers().set(HttpHeaderNames.CONTENT_TYPE, contentType);
+        httpResponse.headers().set(HttpHeaderNames.CONTENT_LENGTH, body.length());
+        addCORSHeaders(httpResponse);
+        return httpResponse;
+    }
+
+    public static FullHttpResponse buildJsonErrorResponse(Throwable err) {
+        while (err instanceof CompletionException) {
+            err = err.getCause();
+        }
```

Comment:
```suggestion
    public static FullHttpResponse buildJsonErrorResponse(Throwable throwable) {
        Throwable err = throwable;
        while (err instanceof CompletionException) {
            err = err.getCause();
        }
```

[Codacy] Use temporary local variable to avoid reassigning values to incoming parameters of a method.